frodo.js

```diff
@@ -3,49 +3,46 @@
  *
```

Comment:
SCRIPT-I'm impressed with your use of design patterns here, it really enhances the code organization.